src/collision.cpp

```diff
@@ -614,3 +636,69 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 
 	return result;
 }
+
+bool collisionInCollision(Environment *env, IGameDef *gamedef,
+		const aabb3f &box_0, const v3f &pos_f, ActiveObject *self,
+		bool collideWithObjects)
+{
+	#define PROFILER_NAME(text) (s_env ? ("Server: " text) : ("Client: " text))
+	Map *map = &env->getMap();
+	ServerEnvironment *s_env = dynamic_cast<ServerEnvironment*>(env);
+
+	ScopeProfiler sp(g_profiler, PROFILER_NAME("collisionInCollision()"), SPT_AVG);
+
+	/*
+		Collect node boxes
+	*/
+	std::vector<NearbyCollisionInfo> cinfo;
+	{
+	//TimeTaker tt2("collisionMoveSimple collect boxes");
+	ScopeProfiler sp2(g_profiler, PROFILER_NAME("collision collect boxes"), SPT_AVG);
+
+	v3s16 min = floatToInt(pos_f + box_0.MinEdge, BS) - v3s16(1, 1, 1);
+	v3s16 max = floatToInt(pos_f + box_0.MaxEdge, BS) + v3s16(1, 1, 1);
+
+	bool any_position_valid = addAreaNodeBoxes(min, max, gamedef, map, cinfo);
+
+	// Do not move if world has not loaded yet, since custom node boxes
+	// are not available for collision detection.
+	// This also intentionally occurs in the case of the object being positioned
+	// solely on loaded CONTENT_IGNORE nodes, no matter where they come from.
+	if (!any_position_valid) {
+		return true;
+	}
+
+	} // tt2
+
+	if(collideWithObjects)
+	{
+		/* add object boxes to cinfo */
+		v3f speed;
+		addObjectBoxes(env, box_0, 0, pos_f, speed, self, cinfo);
+	} //tt3
+
+	/*
+		Collision detection
+	*/
+
+	{
+		aabb3f checkbox = box_0;
+		checkbox.MinEdge += pos_f;
+		checkbox.MaxEdge += pos_f;
+
+		/*
+			Go through every nodebox, find nearest collision
+		*/
+		for (u32 boxindex = 0; boxindex < cinfo.size(); boxindex++) {
+			const NearbyCollisionInfo &box_info = cinfo[boxindex];
+
+			if (collisionEngeInBox(checkbox.MinEdge, box_info.box) ||
```

Comment:
The purpose of this seems to be to check whether `checkbox` and `box_info` collide? If so, then this would be incorrect, since it just checks min/max edges. There are many cases that doesn't cover (including cases where boxes intersect despite no corner of either box being inside the other box). You should then simply replace this with [`checkbox.intersectsWithBox(box_info.box)`](https://irrlicht.sourceforge.io/docu/classirr_1_1core_1_1aabbox3d.html#a5d3d9d09354c4c362625dab7a15dcedb`).

Reply:
changed